Acceptance test for static tags

sharidas opened the pull request:
Acceptance test added for static tags

Signed-off-by: Sujith H <sharidasan@owncloud.com>

<!--
Thanks for submitting a change to ownCloud!

This is the bug tracker for the Server component. Find other components at https://github.com/owncloud/core/blob/master/.github/CONTRIBUTING.md#guidelines

For fixing potential security issues please see https://owncloud.org/security/

To make it possible for us to get your change reviewed and merged please carefully fill out the requested information below.

Please note that any kind of change needs first be submitted to the master branch which holds the next major version of ownCloud.

We will carefully discuss if your change can or has to be backported to stable branches.

Please set the following labels:

- Set label "3 - To review" for review or "2 - Development" in case the PR still has open tasks
- Set label "backport-request" if backport is needed
- Assignment: assign to self
- Milestone: set the same as the ticket this PR fixes, or "development" by default
- Reviewers: pick at least one
-->

## Description
<!--- Describe your changes in detail -->
Add acceptance test for the static tags

## Related Issue
<!--- This project only accepts pull requests related to open issues -->
<!--- If suggesting a new feature or change, please discuss it in an issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here: -->
- Fixes https://github.com/owncloud/core/issues/33443

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
Acceptance test were not added for the static tags added recently. So this patch is adding acceptance test and hence changes required for webdav. 

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->
- test environment:
- test case 1:
- test case 2:
- ...

## Screenshots (if appropriate):

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:
<!-- Tick the checkboxes when done. -->
<!-- Raise documentation ticket in https://github.com/owncloud/documentation -->
- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:
<!-- In case of incomplete PR, please list the open tasks here -->
- [ ] Backport (if applicable set "backport-request" label and remove when the backport was done)

sharidas committed:
Acceptance test for static tags

Acceptance test added for static tags

Signed-off-by: Sujith H <sharidasan@owncloud.com>

sharidas commented:
Context for this change:
A new tag is added, known as static tags. The acceptance test were not added/updated when the static tags work was done. Hence this PR includes the changes to add acceptance test + changes required for the webdav.

Below is the checklist which this PR should address:
- [ ] Add acceptance test for assign for static tags.
- [ ] Add acceptance test for unassign for static tags.
- [ ] Add acceptance test for edit for static tags.
- [ ] Add acceptance test creating static tags.
- [ ] Add acceptance test for deleting static tags.
- [ ] Updates made for WebDAV code.
- [ ] phpunit test updates.
- [ ] webui acceptance test for static tags.

sharidas commented:
Current status of the PR:
- Have added the code mentioned in the checklist https://github.com/owncloud/core/pull/33517#issuecomment-438653000

PVince81 commented:
@sharidas seems this is not only adding tests but also fixing some issues ? would be good to have this reflected in the PR subject as information, else this would get deleted from changelog

PVince81 reviewed (commented):
the code changes that fixes additional issues and their unit tests look good.

would be good if @phil-davis could review the tests part

individual-it commented:
acceptance tests look good to me

sharidas commented:
@phil-davis @individual-it below are the changes made as per the comments

- [x] Updated the scenario with `And the administrator has created a "static" tag with name "StaticTag" and groups "group1"`
- [x] Adjusted the table columns to line up

Another round of review required for the changes updated in the feature file of acceptance tests.

individual-it commented on tests/acceptance/features/apiTags/editTags.feature:
```diff
@@ -25,6 +25,13 @@ Feature: Editing the tags
+    When the user edits the tag with name "StaticTagName" and sets its name to "AnotherTagName" using the WebDAV API
+    Then the following tags should exist for the administrator
+      | StaticTagName | static |
+
```
one bonus empty line

individual-it commented:
acceptance tests look very good to me :+1:

phil-davis reviewed (commented):
I agree, acceptance tests are good now.
@PVince81 or @DeepDiver1975 please approve and merge when you are happy with the rest of the code.

PVince81 reviewed (approved):
:+1:

PVince81 commented:
@sharidas please backport
